Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -686,7 +688,21 @@ protected class ClassWriter : IClassWriter {
 
     protected override void EmitNewArray(Type elmtType, IToken tok, List<Expression> dimensions, bool mustInitialize,
       ConcreteSyntaxTree wr, ConcreteSyntaxTree wStmts) {
-      throw new NotImplementedException();
+      var initValue = mustInitialize ? DefaultValue(elmtType, wr, tok, true) : null;
+      if (dimensions.Count == 1) {
+        // handle the common case of 1-dimensional arrays separately
+        var arrayType = TypeInitializationValue(elmtType, wr, tok, false, false);
+        wr.Write($"[{initValue} for _ in range");
+        TrParenExpr(dimensions[0], wr, false, wStmts);
+        wr.Write("]");
+      } else {
+        wr.Write("{0}.newArray({1}", DafnyArrayClass, initValue ?? "None");
+        foreach (var dim in dimensions) {
+          wr.Write(",int ");
```

Comment:
```suggestion
          wr.Write(", int");
```
But I think we should just skip those.